File: 04 Largest Palindrome Product/problem4.js
var pallindromeCheck = function(num){
  var string = num.toString();
  return string === string.split('').reverse().join('');
}

var largestPallindromeFinder = function(low, high){
  var largest = 0;
  pallindromes = [];
  for(var i = low; i < high; i ++){
    for(var y = i; y < high; y ++){
      var prod = i * y;
      if ((prod > largest) && (pallindromeCheck(prod))){
        largest = prod;
      }
    }
  }
  return largest;
}

console.log(largestPallindromeFinder(100, 999));

File: 04 Largest Palindrome Product/problem4.swift
func pallindrome_check(num: Int) -> Bool{
  var input = num
  var reversed = 0

  while input > 0 {
    reversed *= 10
    reversed += input % 10
    input /= 10
  }
  return (reversed == num)
}

var largest = 0

for x in 100...999 {
  for y in x...999 {
    var product = x * y
    if ((product > largest) && pallindrome_check(product)) {
      largest = product
    }
  }
}

println(largest)
